packages/material-ui-lab/src/TreeItem/TreeItem.test.js

```diff
@@ -1473,14 +1495,513 @@ describe('<TreeItem />', () => {
+            </TreeView>,
+          );
+
+          act(() => {
+            getByRole('tree').focus();
+          });
+
+          expect(getByTestId('one')).toHaveVirtualFocus();
+
+          fireEvent.keyDown(getByRole('tree'), { key: 'ArrowDown' });
+          expect(getByTestId('three')).toHaveVirtualFocus();
+        });
+
+        it('should not be focused on tree focus', () => {
+          const { getByRole, getByTestId } = render(
+            <TreeView>
+              <TreeItem nodeId="one" label="one" disabled data-testid="one" />
+              <TreeItem nodeId="two" label="two" data-testid="two" />
+            </TreeView>,
+          );
+
+          act(() => {
+            getByRole('tree').focus();
+          });
+
+          expect(getByTestId('two')).toHaveVirtualFocus();
+        });
+      });
+    });
+
+    describe('expansion', () => {
+      describe('`disabledItemsFocusable=true`', () => {
+        it('should prevent expansion on enter', () => {
+          const { getByRole, getByTestId } = render(
+            <TreeView disabledItemsFocusable>
+              <TreeItem nodeId="one" label="one" />
+              <TreeItem nodeId="two" label="two" disabled data-testid="two">
+                <TreeItem nodeId="three" label="three" />
+              </TreeItem>
+            </TreeView>,
+          );
+
+          fireEvent.focusIn(getByTestId('two'));
+          expect(getByTestId('two')).toHaveVirtualFocus();
+          expect(getByTestId('two')).to.have.attribute('aria-expanded', 'false');
+          fireEvent.keyDown(getByRole('tree'), { key: 'Enter' });
+          expect(getByTestId('two')).to.have.attribute('aria-expanded', 'false');
+        });
+
+        it('should prevent expansion on right arrow', () => {
+          const { getByRole, getByTestId } = render(
+            <TreeView disabledItemsFocusable>
+              <TreeItem nodeId="one" label="one" />
+              <TreeItem nodeId="two" label="two" disabled data-testid="two">
+                <TreeItem nodeId="three" label="three" />
+              </TreeItem>
+            </TreeView>,
+          );
+
+          fireEvent.focusIn(getByTestId('two'));
+          expect(getByTestId('two')).toHaveVirtualFocus();
+          expect(getByTestId('two')).to.have.attribute('aria-expanded', 'false');
+          fireEvent.keyDown(getByRole('tree'), { key: 'ArrowRight' });
+          expect(getByTestId('two')).to.have.attribute('aria-expanded', 'false');
+        });
+
+        it('should prevent collapse on left arrow', () => {
+          const { getByRole, getByTestId } = render(
+            <TreeView defaultExpanded={['two']} disabledItemsFocusable>
+              <TreeItem nodeId="one" label="one" />
+              <TreeItem nodeId="two" label="two" disabled data-testid="two">
+                <TreeItem nodeId="three" label="three" />
+              </TreeItem>
+            </TreeView>,
+          );
+
+          fireEvent.focusIn(getByTestId('two'));
```

Comment:
question: Is this the only way to move focus to a treeitem programmatically? `.focus()` should fire a `foucsin` event as well (though the JSDOM fix might not be installed yet).

Reply:
Ideally we’d be using `.focus` But I don’t think the fix was installed last time I tried so it was failing.

Reply:
Should work since https://github.com/mui-org/material-ui/pull/21768:

> Added firing of focusin and focusout when using el.focus() and el.blur(). (trueadm)